Model/MenuRepository.php

```diff
@@ -40,8 +43,10 @@ public function __construct(
         MenuFactory $menuFactory,
         CollectionFactory $menuCollectionFactory,
         MenuSearchResultsInterfaceFactory $menuSearchResults,
+        MenuInterface $menu,
```

Comment:
I am not a fan of injecting `MenuInterface` directly as it is data object - especially in this context where we also have `MenuFactory` (but this is not true for 2 other instances).
Maybe this should be some kind of helper to provide link fields?